lib/src/excel.dart

```diff
@@ -1,36 +1,128 @@
-  void exec() async {
+  /// Waits for Office APIs to be ready and then populates the spreadsheet with
+  /// entries in the [insertionOrderList]
+  static void populate(List insertionOrderList) async {
     await Office.onReady(allowInterop((info) async {
-      await ExcelJS.run(allowInterop(_populate));
+      await ExcelJS.run(allowInterop((context) {
+        // Grabs the current worksheet
+        final sheet = context.workbook.worksheets.getActiveWorksheet();
+
+        // Adds a master table
+        final endAddressInt = _startAddressInt + _tableHeader.length - 1;
+        final tableAddress = 'A1:${String.fromCharCode(endAddressInt)}1';
+        final table = context.workbook.tables.add(tableAddress, true);
+
+        // Adds and formats table header
+        table.getHeaderRowRange()
+          ..values = [_tableHeader]
+          ..format.font.name = _fontName
+          ..format.font.size = _fontSize
+          ..format.font.bold = true
+          ..format.horizontalAlignment = _horizontalAlignment
+          ..format.borders.getItem('EdgeTop').style = _borderStyle
+          ..format.borders.getItem('EdgeBottom').style = _borderStyle;
+
+        // Adds and formats table body
+        table.getDataBodyRange()
+          ..formulas = _generateTableBody(insertionOrderList)
+          ..format.font.name = _fontName
+          ..format.font.size = _fontSize
+          ..format.horizontalAlignment = _horizontalAlignment;
+
+        // Auto-fits all used cells
+        sheet.getUsedRange().getEntireColumn().format.autofitColumns();
+        sheet.getUsedRange().getEntireRow().format.autofitRows();
+
+        // Sets the sheet as active
+        sheet.activate();
+        return context.sync();
+      }));
     }));
   }
 
-  Future<void> _populate(RequestContext context) async {
-    final sheet = context.workbook.worksheets.getActiveWorksheet();
-    final range = sheet.getRange('A1');
-    final values = <List<String>>[
-      ['test test']
-    ];
-    range.values = values;
-    range.format.autofitColumns();
-    return context.sync();
+  static List _generateTableBody(List insertionOrderList) {
+    final tableBody = [];
+
+    for (InsertionOrder io in insertionOrderList) {
+      tableBody.add([
+        io.insertionOrderId,
+        io.advertiserId,
+        io.campaignId,
+        io.displayName,
+        io.entityStatus.toString(),
+        io.updateTime,
+        io.pacing.pacingPeriod.toString(),
+        io.pacing.pacingType.toString(),
+        io.pacing.dailyMaxMicros,
+        io.pacing.dailyMaxImpressions,
+        io.budget.budgetUnit.toString(),
+        io.budget.automationType.toString(),
+        _calculateTotalBudgetAmount(io.budget.budgetSegments),
+        _getStartDate(io.budget.budgetSegments.first),
+        _getEndDate(io.budget.budgetSegments.last),
+      ]);
+    }
+    return tableBody;
+  }
+
+  static String _calculateTotalBudgetAmount(
```

Comment:
I think this is not very relevant. Advertisers are mostly interested in their currently active segment.

Reply:
New issue created #35, will be addressed in a separate PR.

Reply:
ok lets make sure #35 doesnt fall off since its part of our mvp

Reply:
you can drop a todo with the issue on the method